applications/nrf5340_audio/src/modules/sd_card.c

```diff
@@ -180,6 +195,136 @@ int sd_card_read(char const *const filename, char *const data, size_t *size)
 	return 0;
 }
 
+int sd_card_segment_open(char const *const filename, char const *const path_to_file)
+{
+	int ret;
+
+	if (seg_read_started) {
+		LOG_ERR("Segment read has already started");
+		/* Which value should I return here? */
+		return -1;
+	}
+
+	if (strlen(path_to_file) + strlen(abs_path_name) > PATH_MAX_LEN) {
+		LOG_ERR("Filepath is too long");
+		return -EINVAL;
+	}
+
+	char abs_path_name[PATH_MAX_LEN + 1] = SD_ROOT_PATH;
+	strcat(abs_path_name, path_to_file);
+	LOG_INF("abs path name:\t%s", abs_path_name);
+
+	if (!sd_init_success) {
+		return -ENODEV;
+	}
+
+	if (strlen(filename) > CONFIG_FS_FATFS_MAX_LFN) {
+		LOG_ERR("Filename is too long");
+		return -ENAMETOOLONG;
+	}
+
+	strcat(abs_path_name, filename);
+	fs_file_t_init(&f_seg_read_entry);
+
+	ret = fs_open(&f_seg_read_entry, abs_path_name, FS_O_READ);
+	if (ret) {
+		LOG_ERR("Open file failed");
+		return ret;
+	}
+
+	seg_read_started = true;
```

Comment:
We need to discuss here. We cannot have a seg_read_started here. What if another thread wants to open another segment? This approach does not make this thread safe.

Reply:
Will close this after our latest changes have been merged.